apps/gitness/src/pages-v2/repo/stores/repo-branch-store.tsx

```diff
@@ -0,0 +1,80 @@
+import { create } from 'zustand'
+
+import {
+  CalculateCommitDivergenceOkResponse,
+  TypesBranchExtended,
+  TypesRepository
+} from '@harnessio/code-service-client'
+
+import { timeAgoFromISOTime } from '../../../pages/pipeline-edit/utils/time-utils'
+
+interface BranchData {
+  id: number
+  name: string
+  sha: string
+  timestamp: string
+  user: {
+    name: string
+    avatarUrl: string
+  }
+  behindAhead: {
+    behind: number
+    ahead: number
+    default: boolean
+  }
+}
+
+interface BranchStore {
+  // State
+  branches: BranchData[]
+  defaultBranch: string
+  branchDivergence: CalculateCommitDivergenceOkResponse
+
+  // Actions
+  setBranchesData: (
+    branches: TypesBranchExtended[],
+    divergence: CalculateCommitDivergenceOkResponse,
+    defaultBranch?: string
+  ) => void
+  setDefaultBranch: (metadata: TypesRepository) => void
+  setBranchDivergence: (divergence: CalculateCommitDivergenceOkResponse) => void
+}
+
+export const useRepoBranchStore = create<BranchStore>(set => ({
+  // Initial state
+  branches: [],
+  defaultBranch: '',
```

Comment:
we should also track the currently selected branch

Reply:
Sure, lets do this in a separate PR though. Let this one be focused on just the `branch-list` page and making sure the merged store works across all pages for now?